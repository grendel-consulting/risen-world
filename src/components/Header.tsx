/** @jsx jsx */
import React from "react";
import { graphql, useStaticQuery, Link } from "gatsby";
import { jsx, Box, Container } from "theme-ui";

export type HeaderProps = {
  // empty
};

export const Header: React.FC<HeaderProps> = (props) => {
  const data = useStaticQuery(graphql`
    query {
      site {
        siteMetadata {
          title
        }
      }
    }
  `);

  return (
    <header>
      <Box bg="primary" color="background" paddingY={2}>
        <Container>
          <h1 sx={{ margin: 0, fontSize: 4 }}>
            <Link to="/" sx={{ color: `inherit` }}>
              {data.site.siteMetadata.title}
            </Link>
          </h1>
        </Container>
      </Box>
    </header>
  );
};
export default Header;
