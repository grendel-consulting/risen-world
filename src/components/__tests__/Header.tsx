import React from "react";
import renderer from "react-test-renderer";

import { PureHeader as Header } from "../Header";

describe(`Header`, () => {
  it(`renders correctly`, () => {
    const data = {
      site: {
        siteMetadata: {
          title: `Risen World`,
        },
      },
    };

    const tree = renderer.create(<Header data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
