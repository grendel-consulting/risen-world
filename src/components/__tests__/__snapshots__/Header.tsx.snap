// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <div
    className="css-btmb2w"
  >
    <div
      className="css-1yrmf69"
    >
      <h1
        className="css-1eq3ttl-PureHeader"
      >
        <a
          className="css-1uop0s9-PureHeader"
          href="/"
        >
          Risen World
        </a>
      </h1>
    </div>
  </div>
</header>
`;
